packages/mask/src/components/shared/SelectRecipients/SelectRecipientsDialog.tsx

```diff
@@ -24,72 +81,116 @@ export interface SelectRecipientsDialogUIProps extends withClasses<never> {
     disabled: boolean
     disabledItems?: Profile[]
     submitDisabled: boolean
+    loading?: boolean
+    searchEmptyText?: string
     onSubmit: () => void
     onClose: () => void
     onSelect: (item: Profile) => void
     onDeselect: (item: Profile) => void
+    onSearch(v: string): void
 }
 export function SelectRecipientsDialogUI(props: SelectRecipientsDialogUIProps) {
     const { t } = useI18N()
-    const classes = useStylesExtends(useStyles(), props)
-    const { items, disabledItems } = props
+    const { classes, cx } = useStyles()
+    const { items, disabledItems, onSearch } = props
     const [search, setSearch] = useState('')
+    const [, copyToClipboard] = useCopyToClipboard()
+    const copyFingerprint = useSnackbarCallback({
+        executor: async (v: string) => copyToClipboard(v),
+        deps: [],
+        successText: 'public key copied',
+    })
+    useMemo(() => {
+        setSearch('')
+        onSearch('')
+    }, [props.open])
     const itemsAfterSearch = useMemo(() => {
         const fuse = new Fuse(items, {
-            keys: ['identifier.userId', 'fingerprint', 'nickname'],
+            keys: ['identifier.userId', 'nickname', 'address', 'publicHexKey'],
             isCaseSensitive: false,
             ignoreLocation: true,
             threshold: 0,
         })
-
-        return search === '' ? items : fuse.search(search).map((item) => item.item)
+        return items
```

Comment:
Why skip searching?

By the way, I think we should memoize fuse individually, since the keyword `search` might change more frequently.

```ts
const fuse = useMemo(() =>  new Fuse(items, {}), [items])

const searchedItems = useMemo(() => {
   return search === '' ? items : fuse.search(search).map((item) => item.item)
}, [fuse, search])
```

Reply:
thats a mistake

Reply:
reverted